xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-test/xwiki-platform-ckeditor-test-docker/src/test/it/org/xwiki/ckeditor/test/ui/FilterIT.java

```diff
@@ -88,4 +89,26 @@ void escapeStyleContent()
         // Verify that the origial style content is preserved.
         assertSourceContains(source.toString() + " end");
     }
+
+    @Test
+    @Order(2)
+    void listUnbreakableSpaceMerge()
+    {
+        // See https://jira.xwiki.org/browse/XWIKI-22024.
+        this.editor.getToolBar().clickNumberedList();
+
+        // Write the triggering text: any text between round brackets, followed by more text, a space and more text
+        // again.
+        this.textArea.sendKeys("(test)s test");
+
+        // Move to inside the round brackets and change the content inside the bracket.
+        for (int i = 0; i < 7; i++)
+            this.textArea.sendKeys(Keys.ARROW_LEFT);
```

Comment:
I would call ``sendKeys`` a single time, by repeating the left arrow (using ``StringUtils`` for instance).